.github/workflows/slither.yml

```diff
@@ -0,0 +1,38 @@
+name: Static Anaysis with Slither
+
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Install solc-select
+        run: pip3 install solc-select
+
+      - name: Install solc 0.8.16
+        run: |
+          solc-select install 0.8.16
+          solc-select use 0.8.16
```

Comment:
I wish we could reduce the need to introduce Python dependencies for our dev tooling. Would prefer to keep things in TypeScript and Rust.

Reply:
pip3 is used to install solc-select as a slither dependency. It needs to be there